### PR TITLE
ls: remove all globals

### PR DESCRIPTION
**Pull request by rminnich**
This makes it usable in tamago and other bare-metal environments.

### CODE REVIEW

File: cmds/core/ls/ls.go [hunk 1]
```diff
@@ -34,16 +34,17 @@ import (
...
-	all       = flag.BoolP("all", "a", false, "show hidden files")
-	human     = flag.BoolP("human-readable", "h", false, "human readable sizes")
-	directory = flag.BoolP("directory", "d", false, "list directories but not their contents")
-	long      = flag.BoolP("long", "l", false, "long form")
-	quoted    = flag.BoolP("quote-name", "Q", false, "quoted")
-	recurse   = flag.BoolP("recursive", "R", false, "equivalent to findutil's find")
-	classify  = flag.BoolP("classify", "F", false, "append indicator (one of */=>@|) to entries")
-	size      = flag.BoolP("size", "S", false, "sort by size")
-)
+type cmd struct {
```
Comment (10000TB) [inline cmds/core/ls/ls.go:37]: the comment for available options are slightly out of sync.

File: cmds/core/ls/ls_plan9.go [hunk 2]
```diff
@@ -18,18 +18,18 @@ import (
...
+func (c cmd) printFile(w io.Writer, stringer ls.Stringer, f file) {
 	if f.err != nil {
 		fmt.Fprintln(w, f.err)
 		return
 	}
 	// Hide .files unless -a was given
-	if *all || !strings.HasPrefix(f.lsfi.Name, ".") {
+	if c.all || !strings.HasPrefix(f.lsfi.Name, ".") {
 		// Unless they said -p, we always print the full path
 		if !*final {
```
Comment (binjip978) [inline cmds/core/ls/ls_plan9.go:29]: The last global, but it should not be a problem for tamago.

**Review by binjip978 — DISMISSED** [review]

File: cmds/core/ls/ls_plan9.go (hunk 2 above)
Comment (rminnich) [reply cmds/core/ls/ls_plan9.go:28]: yeah, I don't know how long I care about Plan 9, but we'll see. We might actually want this option for the standard ls anyway ...

File: cmds/core/ls/ls.go (hunk 1 above)
Comment (rminnich) [reply cmds/core/ls/ls.go:41]: I don't know. it was a convention when we started this project, but if you think it makes sense to remove them, I would agree.

**Review by binjip978 — APPROVED** [review]

**Pull request merged**